src/nvim/option.c

```diff
@@ -3071,6 +3071,13 @@ did_set_string_option(
     if (!parse_winhl_opt(curwin)) {
       errmsg = e_invarg;
     }
+  } else if (varp == &p_tst) {  // 'treesitter'
```

Comment:
Related:

* https://github.com/neovim/neovim/pull/16008#issuecomment-945235165
* https://github.com/neovim/neovim/pull/15260#issuecomment-945235576
* In https://github.com/neovim/neovim/pull/15585 we used `vim.diagnostic.config` for diagnostics options.
    * Ideally, "vim options" are probably the Right Way. So +1 for this 'treesitter' option.